src/checkbox/tests/unit/Checkbox.ts

```diff
@@ -1,151 +1,230 @@
-		});
-		const vnode = <VNode> checkbox.__render__();
-		const labelNode = vnode.children![0];
-		const inputNode = vnode.children![1].children![0];
-
-		assert.isTrue(inputNode.properties!.checked);
-		assert.strictEqual(inputNode.properties!['aria-describedby'], 'id1');
-		assert.isTrue(inputNode.properties!.disabled);
-		assert.strictEqual(inputNode.properties!['aria-invalid'], 'true');
-		assert.strictEqual(inputNode.properties!.name, 'bar');
-		assert.isTrue(inputNode.properties!.readOnly);
-		assert.strictEqual(inputNode.properties!['aria-readonly'], 'true');
-		assert.isTrue(inputNode.properties!.required);
-		assert.strictEqual(inputNode.properties!.value, 'qux');
-
-		assert.strictEqual(vnode.properties!['form'], 'id2');
-		assert.strictEqual(labelNode.properties!.innerHTML, 'foo');
-	},
+		};
+		const children: HNode[] = [
+			v('div', {
+				classes: widget.classes(css.inputWrapper)
+			}, [
+				v('div', { classes: widget.classes(css.onLabel) }, [ checkboxProperties.onLabel! ]),
+				v('div', { classes: widget.classes(css.offLabel) }, [ checkboxProperties.offLabel! ]),
+				v('input', {
+					'aria-describedby': checkboxProperties.describedBy,
+					'aria-invalid': String(checkboxProperties.invalid),
+					'aria-readonly': String(checkboxProperties.readOnly),
+					bind: widget,
+					checked: checkboxProperties.checked,
+					classes: widget.classes(css.input),
+					disabled: checkboxProperties.disabled,
+					name: checkboxProperties.name,
+					onblur: widget.listener,
+					onchange: widget.listener,
+					onclick: widget.listener,
+					onfocus: widget.listener,
+					onmousedown: widget.listener,
+					onmouseup: widget.listener,
+					ontouchstart: widget.listener,
+					ontouchend: widget.listener,
+					ontouchcancel: widget.listener,
+					readOnly: checkboxProperties.readOnly,
+					required: checkboxProperties.required,
+					type: 'checkbox',
+					value: checkboxProperties.value
+				})
+			])
+		];
+
+		widget.setProperties(checkboxProperties);
 
-	'state classes'() {
+		// TODO: is there a better way to do this than a throwaway Checkbox instance?
 		const checkbox = new Checkbox();
-		checkbox.setProperties({
-			checked: true,
-			disabled: true,
-			invalid: true,
-			readOnly: true,
-			required: true,
-			mode: Mode.toggle,
-			offLabel: 'Off',
-			onLabel: 'On'
-		});
-		let vnode = <VNode> checkbox.__render__();
-
-		assert.isTrue(vnode.properties!.classes![css.checked]);
-		assert.isTrue(vnode.properties!.classes![css.disabled]);
-		assert.isTrue(vnode.properties!.classes![css.invalid]);
-		assert.isTrue(vnode.properties!.classes![css.readonly]);
-		assert.isTrue(vnode.properties!.classes![css.required]);
-		assert.isTrue(vnode.properties!.classes![css.toggle]);
-
-		checkbox.setProperties({
-			checked: false,
-			disabled: false,
+		widget.expectRender(w<LabelProperties>(Label, {
+			classes: checkbox.classes(css.root, css.toggle, css.checked, css.disabled, css.invalid, css.readonly, css.required),
```

Comment:
Themeable has changed a little since it was written. Now could we just do 
```javascript
w(Label, {
   overrideClasses: { root: css.root css.toggle ...etc }
})
```
in form widgets, and take out the `classes` property on Label entirely?